[nw_iranian_latin] Removed Balochi lang. codes. Updated copyright dates and changed version to 1.0.1

rmlockwood opened the pull request:
…ion to 1.0.1

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

LornaSIL commented:
@rmlockwood I think you will give a better user experience if you include the ISO 15924 script codes. 
So, for the keyboard info file, change: 

```
    "languages": [
        "avd",
        "atn",
```
etc
to
```
    "languages": [
        "avd-Latn",
        "atn-Latn",
```
etc

Then, you should add all the BCP 47 codes to the kps file as well (on the *Keyboards* tab, *Languages* section. Make sure to indicate Latn in the code too as I suspect "Arab" will be the default it will offer.

I believe this is right: The .kps puts the language names into the .kmp file and is used on installation. The keyboard_info file is used when searching on the website.

Not a showstopper, but I would prefer your copyright dates to include the full year > "2017-2019" instead of "2017-19".

rmlockwood commented:
Hi Lorna,

 

I just wanted to do a quick update. Things are never as quick as you think. I guess I can make those changes.

 

Before I do that. When I look at the help for the NW Iranian keyboard I see that there already is a version 1.0.1 done back in June 2018. The source files I edited had a date of June 2017 and only version 1.0. Did something get lost? Can we recover that before I make more changes?

 

Thanks,

Ron

 

From: Lorna Evans <notifications@github.com> 
Sent: Monday, February 25, 2019 5:24 PM
To: keymanapp/keyboards <keyboards@noreply.github.com>
Cc: R Lockwood <rlbooks@web.de>; Mention <mention@noreply.github.com>
Subject: Re: [keymanapp/keyboards] Removed Balochi lang. codes. Updated copyright dates and changed vers… (#620)

 

@rmlockwood <https://github.com/rmlockwood>  I think you will give a better user experience if you include the ISO 15924 script codes.
So, for the keyboard info file, change:

    "languages": [
        "avd",
        "atn",

etc
to

    "languages": [
        "avd-Latn",
        "atn-Latn",

etc

Then, you should add all the BCP 47 codes to the kps file as well (on the Keyboards tab, Languages section. Make sure to indicate Latn in the code too as I suspect "Arab" will be the default it will offer.

I believe this is right: The .kps puts the language names into the .kmp file and is used on installation. The keyboard_info file is used when searching on the website.

Not a showstopper, but I would prefer your copyright dates to include the full year > "2017-2019" instead of "2017-19".

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/keymanapp/keyboards/pull/620#issuecomment-467076612> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AGwoCfelooRvR_6zCby1FjCw66xtbXHZks5vRA4pgaJpZM4bQKCq> .

LornaSIL commented:
It looks like a touch layout was added as well as some infrastructure changes. I think it might be best if you close this PR. Pull from the main repo so all your files are up to date and then make the changes again in a new PR. Sorry :-( (Please do add Script codes like I suggested, also I saw you turned off <CheckFilenameConventions>False</CheckFilenameConventions>. I don't recommend doing that. Try to keep with sticking to the recommended conventions with file naming.

Here's the link to the issue that made the changes:
https://github.com/keymanapp/keyboards/pull/349/commits/d99377537bc4cc3410575301958a0cb2c01d84e7

rmlockwood commented:
I did pull from the main repo before I started. Can you try pulling from there and see if you also don’t see the 1.0.1 with the touch keyboard? I don’t understand, ‘you turned off False’?

 

Thanks,

Ron

 

From: Lorna Evans <notifications@github.com> 
Sent: Tuesday, February 26, 2019 6:56 PM
To: keymanapp/keyboards <keyboards@noreply.github.com>
Cc: R Lockwood <rlbooks@web.de>; Mention <mention@noreply.github.com>
Subject: Re: [keymanapp/keyboards] Removed Balochi lang. codes. Updated copyright dates and changed vers… (#620)

 

It looks like a touch layout was added as well as some infrastructure changes. I think it might be best if you close this PR. Pull from the main repo so all your files are up to date and then make the changes again in a new PR. Sorry :-( (Please do add Script codes like I suggested, also I saw you turned off False. I don't recommend doing that. Try to keep with sticking to the recommended conventions with file naming.

Here's the link to the issue that made the changes:
 <https://github.com/keymanapp/keyboards/commit/d99377537bc4cc3410575301958a0cb2c01d84e7> d993775

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub <https://github.com/keymanapp/keyboards/pull/620#issuecomment-467544248> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AGwoCaOmi7xwEu2SiI2PqOmKbbg0XwYpks5vRXVGgaJpZM4bQKCq> .

LornaSIL commented:
>Can you try pulling from there and see if you also don’t see the 1.0.1 with the touch keyboard?

Yes, the main repo has a current version of 1.0.1 with a touch layout. And your PR has the "new" version as being 1.0.1. There are also merge conflicts. When I see the comparison of your files to what is in your master it looks like you started with 1.0 files. (See the commits link at the top of the page with your first commit. You can see the comparison of differences in the files.) I'm not sure what to do except start over. @mcdurdin do you have advice?

> I don’t understand, ‘you turned off False’?
I guess this is a new setting in Keyman 11. With .kmn open, go to "Project / Project Setting" and make sure "Check filename conventions" is checked.

mcdurdin commented:
In terms of pulling the changes, you should do something like the following:

```
git checkout master
git fetch upstream
git merge upstream/master
git push origin master
git checkout nw_iranian_latin
git merge master
```

This last step will likely result in merge conflicts on:

`release/n/nw_iranian_latin/LICENSE.md`
`release/n/nw_iranian_latin/README.md`
`release/n/nw_iranian_latin/nw_iranian_latin.keyboard_info`
`release/n/nw_iranian_latin/nw_iranian_latin.kpj`
`release/n/nw_iranian_latin/source/help/nw_iranian_latin.php`
`release/n/nw_iranian_latin/source/nw_iranian_latin.kmn`
`release/n/nw_iranian_latin/source/nw_iranian_latin.kps`
`release/n/nw_iranian_latin/source/welcome/welcome.htm`

so you may find it easier to start again as Lorna says, and reapply your changes:

```
git checkout master
git fetch upstream
git merge upstream/master
git push origin master
git checkout -b nw_iranian_latin-2
```
... apply your changes, and push and create Pull Request.

darcywong00 commented on release/n/nw_iranian_latin/nw_iranian_latin.keyboard_info:
```diff
@@ -71,8 +68,8 @@
     ],
-    "version": "1.0",
-    "minKeymanVersion": "9.0",
+    "version": "1.0.1",
```
I've wondered why this keyboard shows up in the "Undetermined" language when looking for the SIL IPA keyboard.
I see its' cause line 22
` "und-fonipa",`

darcywong00 replied:
Just confirming in case we get to improving the user search for "IPA" that you want this keyboard included.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

keyman-server commented:
Keyboards: This pull request is not from a keyboards maintainer and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.
